unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -118,38 +124,42 @@ private async UniTask LoadAndDisplayEquippedWearablesAsync(UserProfile userProfi
         private void LoadAndShowOwnedWearables(UserProfile userProfile)
         {
             view.SetCollectibleWearablesLoadingActive(true);
-            wearableCatalogBridge.RequestOwnedWearables(userProfile.userId)
-                                 .Then(wearables =>
-                                  {
-                                      string[] wearableIds = wearables.GroupBy(i => i.id).Select(g => g.First().id).Take(MAX_NFT_COUNT).ToArray();
-                                      userProfile.SetInventory(wearableIds);
-                                      loadedWearables.AddRange(wearableIds);
-                                      var containedWearables = wearables.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT)
-                                         .Where(wearable => wearableCatalogBridge.IsValidWearable(wearable.id));
-                                      view.SetCollectibleWearables(containedWearables.ToArray());
-                                      view.SetCollectibleWearablesLoadingActive(false);
-                                  })
-                                 .Catch(Debug.LogError);
+            wearablesPromise = wearableCatalogBridge.RequestOwnedWearables(userProfile.userId).Then(wearables =>
+            {
+                string[] wearableIds = wearables.GroupBy(i => i.id).Select(g => g.First().id).Take(MAX_NFT_COUNT).ToArray();
+                userProfile.SetInventory(wearableIds);
+                loadedWearables.AddRange(wearableIds);
+
+                var containedWearables = wearables.GroupBy(i => i.id)
+                                                  .Select(g => g.First())
+                                                  .Take(MAX_NFT_COUNT)
+                                                  .Where(wearable => wearableCatalogBridge.IsValidWearable(wearable.id));
+
+                view.SetCollectibleWearables(containedWearables.ToArray());
+                view.SetCollectibleWearablesLoadingActive(false);
+            });
+
+            wearablesPromise.Catch(Debug.LogError);
         }
 
         private void LoadAndShowOwnedEmotes(UserProfile userProfile)
         {
             view.SetCollectibleEmotesLoadingActive(true);
-            emotesCatalogService.RequestOwnedEmotes(userProfile.userId)
-                                 .Then(emotes =>
-                                  {
-                                      WearableItem[] emoteItems = emotes.GroupBy(i => i.id).Select(g => g.First()).Take(MAX_NFT_COUNT).ToArray();
-                                      view.SetCollectibleEmotes(emoteItems);
-                                      view.SetCollectibleEmotesLoadingActive(false);
-                                  })
-                                 .Catch(Debug.LogError);
+            emotesPromise = emotesCatalogService.RequestOwnedEmotes(userProfile.userId)
```

Comment:
Didnt see this in the previous review, but we should avoid as possible the promises in this service use instead the nice async wrapper =D